### ISSUE TITLE
Feature request: option to group multiple processes of programs in any view

### ISSUE THREAD

**Issue by bege10** [winsiderss/systeminformer#1941]
### Discussed in https://github.com/winsiderss/systeminformer/discussions/1940

<div type='discussions-op-text'>

<sup>Originally posted by **bege10** December 23, 2023</sup>
I miss the view which the Windows Task Manager shows that multiple processes of a program (e.g. Firefox) are grouped together and can be (un-)collapsed. </div>

My requests are:

- Option to collapse by default for all processes
- Keep that view with any sorting (by name, CPU usage ...), not only in tree view.

**Comment by fbck** [comment]
yep, that'd really help to grasp the amount of resources (cpu, ram, etc) a process group is using when sorting

**Comment by kotenok2000** [comment]
You mean like this?
Name column has three sorting states.
![image](https://github.com/winsiderss/systeminformer/assets/8822292/b790c96d-47ac-40c5-8545-b5b719fbf635)

**Comment by bege10** [comment]
> You mean like this? Name column has three sorting states. 
This is what is requested for sorting by any column (see OP).

**Comment by jxy-s** [comment]
Related change (https://github.com/winsiderss/systeminformer/issues/2556#issuecomment-3042293180) for sorting children in the tree - coverts part of the ask. I think the other is mostly already available with "Include usage of collapsed processes". Showing grouped processes (like Task Manager) is probably best suited for its own tab, distinct from the processes tab.

Maybe with https://github.com/winsiderss/systeminformer/issues/2556#issuecomment-3042293180 and "Include usage of collapsed processes" the ask here is mostly satisfied?

**Comment by jxy-s** [comment]
I've made some adjustments that I believe will satisfy the request here.

First there are now view options to allow sorting of child and root processes, this presents the process tree in a way where the tree is always visible and allows for full sorting while keeping the tree present:

<img width="574" height="401" alt="Image" src="https://github.com/user-attachments/assets/5b9141fa-d94a-48c1-8720-3133fe544dd6" />

Then, when combined with the existing "Include usage of collapsed processes" this will present the collapsed nodes as an aggregate of their respective measurements. Some changes needed to be made to support sorting the tree by their aggregate values.

<img width="558" height="244" alt="Image" src="https://github.com/user-attachments/assets/b9aa83a6-f8dd-4d19-8342-ca15a6bcc138" />

The only part that isn't complete yet is the request to default collapse all nodes by default. But in the meantime I added some additional view options to collapse/expand all nodes (see above).

**Comment by bege10** [comment]
> I've made some adjustments that I believe will satisfy the request here.

In which version will this be implemented?

I see version 3.2.25011.2103 here but version 3.2.25099.1530 in winget. Which version is up to date?

**Comment by vladdeda** [comment]
> > I've made some adjustments that I believe will satisfy the request here.
> 
> In which version will this be implemented?
> 
> I see version 3.2.25011.2103 here but version 3.2.25099.1530 in winget. Which version is up to date?

check for updates under "Help" select the canary build.

**Comment by bege10** [comment]
This is a big progress. Thank you very much.

Where do I find this option?
> "Include usage of collapsed processes"

**Comment by bege10** [comment]
What I still miss is the overall memory use of a process with subprocesses, e.g. Firefox. I see all (sub-)processes when expanded or the main process when collapsed. But I don't see the memory use of the whole program.
